python/ray/experimental/workflow/workflow_context.py

```diff
@@ -0,0 +1,42 @@
+import pathlib
+from typing import Optional
+
+
+class WorkflowContext:
+    def __init__(self, workflow_id=None, workflow_root_dir=None, workflow_scope=None):
+        self.workflow_id = workflow_id
+        self.workflow_root_dir = workflow_root_dir
+        self.workflow_scope = workflow_scope or []
+
+    def __reduce__(self):
+        return WorkflowContext, (self.workflow_id, self.workflow_root_dir, self.workflow_scope)
+
+
+_context: Optional[WorkflowContext] = None
```

Comment:
Does this mean that if within a python script, we can only start one workflow job?

Reply:
For workflow job, yes (but we can start multiple workflow steps though). This context is used for workflow steps. If we need to launch multiple jobs, I can create contexts for workflow jobs.

Reply:
different workflows will run in the same process, so we should support it.

Reply:
This looks more like workflow management to me. I have renamed `WorkflowContext` to `WorkflowStepContext` to clarify its usage. I prefer support this in another PR involving workflow management (start, terminate, querying), so we can move fast.